networkit/cpp/graph/GraphBuilder.cpp

```diff
@@ -239,60 +254,75 @@ void GraphBuilder::toGraphParallel(Graph &G) {
         assert(outDeg <= n);
         // allocate enough memory for all edges/weights
         if (directed) {
-            G.inEdges[v].reserve(inDeg);
-            if (weighted) {
-                G.inEdgeWeights[v].reserve(inDeg);
-            }
-        } else {
-            G.outEdges[v].reserve(outDeg + inDeg);
-            if (weighted) {
-                G.outEdgeWeights[v].reserve(outDeg + inDeg);
-            }
+            G.preallocateDirectedInEdges(v, inDeg);
+            } else {
+            G.preallocateUndirected(v, outDeg + inDeg);
         }
 
         // collect 'second' half of the edges
         if (directed) {
-            for (int tid = 0; tid < maxThreads; tid++) {
-                copyAndClear(inEdgesPerThread[tid][v], G.inEdges[v]);
-            }
             if (weighted) {
                 for (int tid = 0; tid < maxThreads; tid++) {
-                    copyAndClear(inWeightsPerThread[tid][v], G.inEdgeWeights[v]);
+                    for (index j = 0; j < inEdgesPerThread[tid][v].size(); j++){
+                    G.addPartialInEdge(unsafe, v, inEdgesPerThread[tid][v][j], inWeightsPerThread[tid][v][j]);
+                    }
+                }
+            } else {
+                for (int tid = 0; tid < maxThreads; tid++) {
+                    for (index j = 0; j < inEdgesPerThread[tid][v].size(); j++){
+                        G.addPartialInEdge(unsafe, v, inEdgesPerThread[tid][v][j]);
+                    }
```

Comment:
Why do we need a new else case here?

Reply:
It is so we don't try to access the weight array when we don't have any weights, which would cause a segmentation fault

Reply:
Makes sense. How did the old code deal with that?

Reply:
it just replaced both arrays in graph, since it would just put an array of size 0 into in the unweighted case.